frappe/website/page_renderers/redirect_page.py

```diff
@@ -14,7 +14,7 @@ def render(self):
 		return build_response(
 			self.path,
 			"",
-			301,
+			self.http_status_code,
```

Comment:
Thanks a lot for the work! Looks like the Web Forms are broken though. _**Explanation below the picture.**_

https://cloud.cypress.io/projects/92odwv/runs/36759/test-results/1c06f907-48d1-4a44-bc2c-63cfdadae9fb/replay?att=1&ts=1702547912923

![image](https://github.com/frappe/frappe/assets/10946971/ea004278-257e-4477-a7b2-714b240c69bc)

---

I believe that at this point, in some code paths, the `self.http_status_code` property is equal to **200**, because of the following code:

https://github.com/frappe/frappe/blob/40e48c9ac4696b5f0f387535c461a779224c5386/frappe/website/serve.py#L19-L20

If `http_status_code` is not specified, it equals **200** (I'm almost certain it's what's happening), then is passed down as a param to RedirectPage, which now has its `self.http_status_code` property equal to **200**, which then renders a _**200 OK blank page**_ with a Location header. (As seen in the picture above)

Looks like a tough architectural problem to solve though, but it's not your fault directly!

Reply:
Basically, the fallback to 200 happens too early, but we want the caller of `get_response(path=None, http_status_code=200)` to have control over the HTTP status code, except in the case of a redirect where it is ***never*** wanted.

Maybe it possible to have:

```diff
class RedirectPage:
	def __init__(self, path, http_status_code=301):
		self.path = path
		self.http_status_code = http_status_code
+		if self.http_status_code == 200:
+			self.http_status_code = 301
```

But this looks kinda dirty

Reply:
@cogk @akhilnarang  `frappe.Redirect` can specify the redirect code? 

Exceptions are just normal objects, can attach more info there.

Reply:
Rest LGTM. Merge when ready :+1: